apps/permissions.py

```diff
@@ -127,3 +144,30 @@ class IsOwner(permissions.BasePermission):
     def has_object_permission(self, request, view, obj):
         # Permissions are only allowed to the owner of the shipment.
         return has_owner_access(request, obj)
+
+
+class UserHasShipmentPermission(HasViewSetActionPermissions):
```

Comment:
Instead of subclassing HasViewSetActionPermissions I think it may make sense to make a separate permission class to represent "IsShipperCarrierModerator", and then add both of them to the Shipment. But I'm not entirely familiar with how HasViewSetActionPermissions is supposed to work - might should ask Lucas.